airbyte-integrations/connectors/source-mysql-v2/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlSourceOperations.kt

```diff
@@ -0,0 +1,223 @@
+        notInteger: Boolean,
+    ): JdbcFieldType<*> =
+        // This mapping includes literals returned by the JDBC driver as well as
+        // *_TYPE_NAME column values from queries to ALL_* system tables.
+        when (typeName) {
+            "BINARY_FLOAT" -> FloatFieldType
+            "BINARY_DOUBLE" -> DoubleFieldType
+            "FLOAT",
+            "DOUBLE PRECISION",
+            "REAL", -> BigDecimalFieldType
+            "NUMBER",
+            "NUMERIC",
+            "DECIMAL",
+            "DEC", -> if (notInteger) BigDecimalFieldType else BigIntegerFieldType
+            "INTEGER",
+            "INT",
+            "SMALLINT", -> BigIntegerFieldType
+            "BOOLEAN",
+            "BOOL", -> BooleanFieldType
+            "CHAR",
+            "VARCHAR2",
+            "VARCHAR",
+            "CHARACTER",
+            "CHARACTER VARYING",
+            "CHAR VARYING", -> StringFieldType
+            "NCHAR",
+            "NVARCHAR2",
+            "NCHAR VARYING",
+            "NATIONAL CHARACTER VARYING",
+            "NATIONAL CHARACTER",
+            "NATIONAL CHAR VARYING",
+            "NATIONAL CHAR", -> NStringFieldType
+            "BLOB" -> BinaryStreamFieldType
+            "CLOB" -> ClobFieldType
+            "NCLOB" -> NClobFieldType
+            "BFILE" -> BinaryStreamFieldType
+            "DATE" -> LocalDateFieldType
+            "INTERVALDS",
+            "INTERVAL DAY TO SECOND",
+            "INTERVALYM",
+            "INTERVAL YEAR TO MONTH", -> StringFieldType
+            "JSON" -> JsonStringFieldType
+            "LONG",
+            "LONG RAW",
+            "RAW", -> BinaryStreamFieldType
+            "TIMESTAMP",
+            "TIMESTAMP WITH LOCAL TIME ZONE",
+            "TIMESTAMP WITH LOCAL TZ", -> LocalDateTimeFieldType
+            "TIMESTAMP WITH TIME ZONE",
+            "TIMESTAMP WITH TZ", -> OffsetDateTimeFieldType
+            else -> PokemonFieldType
+        }
+
+    override fun generate(ast: SelectQuerySpec): SelectQuery =
+        SelectQuery(ast.sql(), ast.select.columns, ast.bindings())
+
+    fun SelectQuerySpec.sql(): String {
+        val components: List<String> = listOf(select.sql(), from.sql(), where.sql(), orderBy.sql())
+        val sqlWithoutLimit: String = components.filter { it.isNotBlank() }.joinToString(" ")
+        val rownumClause: String =
+            when (limit) {
+                NoLimit -> return sqlWithoutLimit
+                Limit(0) -> "LIMIT 0"
+                is Limit -> "LIMIT ?"
+            }
+        return "$sqlWithoutLimit $rownumClause"
+    }
+
+    fun SelectNode.sql(): String =
+        "SELECT " +
+            when (this) {
+                is SelectColumns -> columns.joinToString(", ") { it.sql() }
+                is SelectColumnMaxValue -> "MAX(${column.sql()})"
+            }
+
+    fun Field.sql(): String = "`$id`"
+
+    fun FromNode.sql(): String =
+        when (this) {
+            NoFrom -> TODO("no such thing in mysql")
```

Comment:
I'm pretty sure you can just return `""` here and everything's going to work fine.